tests/objectives/test_constrained.py

```diff
@@ -0,0 +1,161 @@
+import unittest
+import logging
+import numpy as np
+from simsopt.objectives.functions import Identity, Rosenbrock, Failer, Affine
+from simsopt.objectives.constrained import ConstrainedProblem
+from simsopt._core.util import ObjectiveFailure
+
+#logging.basicConfig(level=logging.DEBUG)
+
+
+class ConstrainedProblemTests(unittest.TestCase):
+
+    def test_bound(self):
+        # single variable
+        iden = Identity()
+        prob = ConstrainedProblem(iden.f, lb=-2.0, ub=np.inf)
+        np.testing.assert_allclose(prob.lb, -2.0, 1e-14)
+        np.testing.assert_allclose(prob.ub, np.inf, 1e-14)
+        self.assertTrue(prob.has_bounds == True)
+        self.assertTrue(prob.has_lc == False)
+        self.assertTrue(prob.has_nlc == False)
+
+        # multivariable, scalar bounds
+        rosen = Rosenbrock()
```

Comment:
Move the bounds specification to the parent classes as suggested.

`rosen = Rosenbrock(upper_bounds=[17.0])`